src/os/bluestore/BlueStore.cc

```diff
@@ -8671,6 +8670,30 @@ int BlueStore::_fsck_on_open(BlueStore::FSCKDepth depth, bool repair)
     dout(1) << __func__ << " debug abort" << dendl;
     goto out_scan;
   }
+
+#ifdef HAVE_LIBZBD
+  if (bdev->is_smr()) {
+    auto a = dynamic_cast<ZonedAllocator*>(shared_alloc.a);
```

Comment:
yeah, i'm not sure how to make this nicer. the bottom line is that the zoned freelist *can't* actually enumerate meaningfully.. it can only make up something with the same dead and unwritten byte count.